scripts/pre-commit.sh

```diff
@@ -82,3 +82,50 @@ if [ $SHOULD_FAIL_PRECOMMIT -ne 0 ]; then
 else
   verify_no_included_apikeys
 fi
+
+# CircleCI doesn't let us split a config file into multiple, so we have a pre-commit hook that does it for us
+# so that the file doesn't become unmanageable. 
+concatenate_circleci_configs() {
```

Comment:
Might be worth extracting this to a separate script and just call it from the `pre-commit.sh`? Specially if we follow my other suggestion about adding a test job to make sure the combined file matches the individual files.